src/foam/nanos/controller/ApplicationController.js

```diff
@@ -626,18 +626,45 @@ foam.CLASS({
       return text;
     },
 
-    function pushMenu(menu, opt_forceReload) {
-      if ( menu.id ) {
-        menu.launch(this);
-        menu = menu.id;
+    async function pushMenu(menu, opt_forceReload) {
+      let idCheck = menu && menu.id ? menu.id : menu;
+      let currentMenuCheck = this.currentMenu && this.currentMenu.id ? this.currentMenu.id : this.currentMenu;
+      // console.log(`menuTryingToPush: ${idCheck}, currentMenu ${currentMenuCheck}, Should I Push? ${currentMenuCheck != idCheck}`);
+      if ( currentMenuCheck == idCheck && ! opt_forceReload ) return;
```

Comment:
Don't we have logic somewhere to load the default menu? Can't we reuse that instead?

Reply:
in Controller.js we have pushDefaultMenu https://github.com/nanoPayinc/NANOPAY/pull/13043/files#diff-96e2487a5eee808b3667210ec68b4efe0070657216d277fc74c4eca30aeab50bR849
but it was doing the same thing as pushMenu ... so it just takes the theme.defaultMenu and calls pushMenu.

So this `if ( currentMenuCheck == idCheck && ! opt_forceReload ) return;` is to stop recursive calls. Will update the documentation here. And its cause the update to the memento and the menu.launch both end up calling pushMenu again. So to stop excess calls this line does its check.

It could be argued that memento.feedback_ is equivalent to this - however that is internal to memento and didn't seem appropriate here.